src/stan/mcmc/chains.hpp

```diff
@@ -701,6 +785,18 @@ namespace stan {
         return effective_sample_size(index(name));
       }
 
+      double effective_sample_size_pod(const int index) const {
+        std::vector<std::vector<double> > samples(num_chains());
+        for (int chain = 0; chain < num_chains(); chain++) {
+          Eigen::VectorXd Esample = this->samples(chain, index);
+          samples[chain].resize(Esample.size());
+          for (int n = 0; n < Esample.size(); n++) {
```

Comment:
This can be done with an iterator copy.

```
std::copy(&Esample[0], &Esample[0] + Esample.size(), samples[chain].begin());
```

But one of the reasons I was suggesting using POD (plain old data) types is that then you don't need this copy, you can just send in the pointers if you use `std::vector<double*>` as the type.  I'm not quite sure what the best thing to do here is.  Given everything else going on, the copies shouldn't be too expensive.
Eventually, we'd want to chase all of this through and change the underlying types of member variables so we don't have to copy.  Is the original result stored in this calss in a `VectorXd`?